pages/index.tsx

```diff
@@ -0,0 +1,584 @@
+      const authorTypeLinkId = await deep.id('@deep-foundation/messaging', 'Author');
+      const messagingTreeId = await deep.id('@deep-foundation/messaging', 'MessagingTree');
+
+      console.log("before insert sound link", sounds);
+      console.log("flakeed2");
+      const { data: [{ id: soundLinkId }] } = await deep.insert(sounds.map((sound) => ({
+        type_id: recordTypeLinkId,
+        in: {
+          data: [{
+            type_id: containTypeLinkId,
+            from_id: audioRecordsLinkId,
+          }]
+        },
+        out: {
+          data: [
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: soundTypeLinkId,
+                  string: { data: { value: sound.record["recordDataBase64"] } },
+                  out: {
+                    data: [
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: mimetypeTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] } },
+                          }
+                        }
+                      },
+                      {
+                        type_id: containTypeLinkId,
+                        to: {
+                          data: {
+                            type_id: formatTypeLinkId,
+                            string: { data: { value: sound.record["mimeType"] === "audio/webm;codecs=opus" ? "webm" : "aac" } },
+                          }
+                        }
+                      }
+                    ]
+                  }
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: durationTypeLinkId,
+                  number: { data: { value: sound.record["msDuration"] } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: startTimeTypeLinkId,
+                  string: { data: { value: sound.startTime } },
+                }
+              }
+            },
+            {
+              type_id: containTypeLinkId,
+              to: {
+                data: {
+                  type_id: endTimeTypeLinkId,
+                  string: { data: { value: sound.endTime } },
+                }
+              }
+            }]
+        }
+      })));
+      console.log("flakeed1");
+      console.log("soundLinkId", soundLinkId)
+
+      const { data: [isGcloudAuthKeyLinkId] } = await deep.select({
+        type_id: gcloudAuthKeyTypeLink,
+        in: {
+          type_id: containTypeLinkId,
+          from_id: deep.linkId,
+        },
+      });
+
+      console.log("googleauth", googleAuth);
+      if (!isGcloudAuthKeyLinkId) {
```

Comment:
We should not even provide ability to the main content page if the required data is not provided.
For example in sdk if you do not pass graphql path and token you just see login page but not main content. Do the same for gcloudAuthKey